src/API/index.js

```diff
@@ -1,87 +1,99 @@
-    errorText = err.response.data;
-  } else if (err.request) {
-    console.log('No response was received', err.request);
-    errorText = 'There was no response from the server.';
-  } else {
-    console.log('Unknown Error', err.message);
-    errorText = err.message;
+/**
+ * URL Maker
+ * @param {string} ext extension to append to url
+ */
+const base_url = `${protocol}://${host}:${port}/api/`;
+
+// Base request method for all endpoints
+async function baseRequest(path, method, body, auth) {
+  let config = {
+    method: method,
+    body: body && JSON.stringify(body),
+    credentials: "include",
+    headers: {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json',
+    },
+  };
+  if (auth) {
+    config.headers.Authorization = `Bearer ${auth}`;
   }
-  return errorText;
-  /* eslint-enable no-console */
+
+  let response = await fetch(base_url + path, config);
+  return response.json();
 }
 
-const routes = {
-  getUser: () => new Promise((resolve, reject) => {
-    axios.get('/api/user')
-      .then((res) => {
-        console.log('user res', res);
-        resolve(res.data);
-      })
-      .catch((err) => {
-        const errorText = errorHandling(err);
-        reject(errorText);
-      });
-  }),
-  setUser: (token, username) => axios.post('/api/user', { token, username }),
-  getAllQuestions: () => new Promise((resolve, reject) => {
-    axios.get('/api/questions')
-      .then((res) => {
-        console.log('Questions', res);
-        resolve(res);
-      })
-      .catch((err) => {
-        const errorText = errorHandling(err);
-        reject(errorText);
-      });
-  }),
-  createQuestion: (question, token) => new Promise((resolve, reject) => {
-    const config = {
-      method: 'get',
-      url: '/api/questions',
+let baseRoutes = {
+  async getDocuments(token) {
+    return baseRequest('document?has_parent=false', 'GET', null, token);
+  },
+
+  async getDocument(doc_id, token) {
+    return baseRequest(`document/${doc_id}`, 'GET', null, token);
+  },
+  async getChildrenByDocument(doc_id, token) {
+    return baseRequest(`document/${doc_id}/children`, 'GET', null, token);
+  },
+
+  async getDocumentData(doc_id, token) {
+    let config = {
+      method: 'GET',
+      credentials: "include",
       headers: {},
-    };
-    if (token) {
-      config.headers.authorization = `Bearer ${token}`;
     }
-    axios.request({
-      method: 'post',
-      url: '/api/questions',
-      data: question,
-    })
-      .then((res) => {
-        resolve(res);
-      })
-      .catch((err) => {
-        const errorText = errorHandling(err);
-        reject(errorText);
-      });
-  }),
-  getQuestions: (token) => new Promise((resolve, reject) => {
-    const config = {
-      method: 'get',
-      url: '/api/questions',
-      headers: {},
-    };
-    if (token) {
-      config.headers.authorization = `Bearer ${token}`;
+    if(token)
+      config.headers.Authorization = `Bearer ${token}`;
```

Comment:
Please fix this syntax. Also make this variable and line 23 consistent. 'Token' makes more sense to me.